src/core/database/cypher.factory.ts

```diff
@@ -41,6 +41,7 @@ export const CypherFactory: FactoryProvider<Connection> = {
       const session: Session | null = origSession.call(conn);
       if (session) {
         const origRun = session.run;
+        // @ts-ignore FIXME: unclear how to get around the TS error here
```

Comment:
Oh gag. This is going to be tricky. Let me think about this more...maybe we just continue to ignore it for now.

Reply:
Sounds good!

Reply:
Ok figured a good step. Just need to return the result of `.call` instead of `.catch`.
```ts
const origRun = session.run;
session.run = function(this: never, origStatement, parameters, conf) {
  const statement = stripIndent(origStatement);
  logger.debug('\n' + statement, parameters);
  const result = origRun.call(session, statement, parameters, conf);
  result.catch(e => {
    throw jestSkipFileInExceptionSource(e, __filename);
  });
  return result;
};
```

Would you mind applying that?

Reply:
Wait now I'm second guessing myself. Would that `catch` callback be applied if it's not returned? Does catch mutate the original promise or return a new one?

Reply:
Now that I'm thinking through it more, I believe `throw` inside of a `catch` will result in an uncaught exception. What if we `await` the result before returning it, capture the error with `try / catch` it then throw inside of the `try / catch` statement?

Reply:
Actually I think this might be okay...testing now.

Reply:
We can't use async await here because the run result is a promise with an additional subscribe method. It's a clever way to support both promise and observable usage. But wrapping it is more annoying.

We don't currently use the observable API so if push comes to shove I'm fine ignoring it. I would like the logic inside the catch to be applied for clarity with errors in jest output.

Reply:
I applied your update, intentionally broke a `session.run` query and tested it, and I'm not seeing any issues with the catches, so I think we should be okay! TS Errors are gone. :D